ArcGISRuntimeSDKQt_QMLSamples/Routing/NavigateRoute/NavigateRoute.qml

```diff
@@ -0,0 +1,277 @@
+                }
+                symbol: stopSymbol
+            }
+            Graphic {
+                Point {
+                    id: aerospaceMuseumPoint
+                    x: -117.147230
+                    y: 32.730467
+                    SpatialReference {wkid: 4326}
+                }
+                symbol: stopSymbol
+            }
+        }
+
+        SimpleMarkerSymbol {
+            id: stopSymbol
+            style: "SimpleMarkerSymbolStyleDiamond"
+            color: "red"
+            size: 20
+        }
+
+        Stop {
+            id: stop1
+            geometry: conventionCenterPoint
+        }
+        Stop {
+            id: stop2
+            geometry: memorialPoint
+        }
+        Stop {
+            id: stop3
+            geometry: aerospaceMuseumPoint
+        }
+
+        RouteTask {
+            id: routeTask
+            url: routeTaskUrl
+            Component.onCompleted: {
+                load();
+            }
+            onLoadStatusChanged: {
+                if (loadStatus === Enums.LoadStatusLoaded) {
+                    createDefaultParameters();
+                }
+            }
+            onCreateDefaultParametersStatusChanged: {
+                if (createDefaultParametersStatus === Enums.TaskStatusCompleted) {
+                    createDefaultParametersResult.returnStops = true;
+                    createDefaultParametersResult.returnDirections = true;
+                    createDefaultParametersResult.returnRoutes = true;
+                    createDefaultParametersResult.outputSpatialReference = SpatialReference.createWgs84();
+                    createDefaultParametersResult.setStops([stop1, stop2, stop3]);
+
+                    //solve the route with these parameters
+                    routeTask.solveRoute(createDefaultParametersResult);
+                }
+            }
+            onSolveRouteStatusChanged: {
+                if (solveRouteStatus === Enums.TaskStatusCompleted) {
+                    if (solveRouteResult.routes.length > 0) {
+                        m_routeResult = solveRouteResult;
+                        m_route = solveRouteResult.routes[0];
+                        mapView.setViewpointGeometryAndPadding(m_route.routeGeometry, 100);
+                        routeAheadGraphic.geometry = m_route.routeGeometry;
+                        navigateButton.enabled = true;
+                    }
+                }
+            }
+        }
+
+        locationDisplay.onLocationChanged: {
+            routeTracker.trackRuntimeLocation(locationDisplay.location);
+        }
+
+        // enable "recenter" button
+        locationDisplay.onAutoPanModeChanged: {
+            recenterButton.enabled = locationDisplay.autoPanMode !== Enums.LocationDisplayAutoPanModeNavigation;
+        }
+
+        Rectangle {
+            id: backBox
+            z: 1
+            width: Qt.platform.os === "ios" || Qt.platform.os === "android" ? 200 : 300
```

Comment:
Although this particular case is probably fine, this sort of thing feel like you're asking for trouble with unusual screen dimensions. Could you use a percentage of the overall screen width instead, or maybe the size of the contents?